test/hypothesis_test.py

```diff
@@ -0,0 +1,38 @@
+"""Property-based tests for Modbus payload round-trip encoding/decoding with randomized word order."""
+import pytest
+from hypothesis import given, settings
+from hypothesis import strategies as st
+
+from pymodbus.client.mixin import ModbusClientMixin as mixin
+
+
+word_order_strategy = st.one_of(st.just("big"), st.just("little"))
```

Comment:
similar to `@pytest.mark.parameterize` but it will mix in the big/little with testing a range of values.